tests/test_query_extensions.py

```
@@ -106,10 +118,148 @@ def test_relevance_score_in_fts_results():
+         patch("mcp_server.database.close_connection"):
+        result = query_email_database(
+            semantic_query="project update",
+            category_filter="finance",
+            project_filter="alpha",
+            limit=10,
+        )
+
+    assert result["results"] == [], f"Expected empty semantic result set from fake cursor, got {result}"
+    assert len(fake_cursor.calls) == 1, f"Expected exactly one semantic query execution, got {fake_cursor.calls}"
+    sql, params = fake_cursor.calls[0]
+    assert "category_tags" in sql and "project_tags" in sql, f"Expected semantic SQL to reuse filter WHERE clauses, got {sql}"
```

Ruff (0.15.15)

[warning] Assertion should be broken down into multiple parts

(PT018)